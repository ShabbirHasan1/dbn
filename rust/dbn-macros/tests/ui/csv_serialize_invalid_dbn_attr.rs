use dbn_macros::CsvSerialize;

#[derive(CsvSerialize)]
#[repr(C)]
struct Record {
    #[dbn(unknown)]
    pub a: u8,
}

fn main() {}
