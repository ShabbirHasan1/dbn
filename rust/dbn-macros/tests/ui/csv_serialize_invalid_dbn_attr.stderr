error: unrecognized dbn attr argument unknown
 --> tests/ui/csv_serialize_invalid_dbn_attr.rs:6:11
  |
6 |     #[dbn(unknown)]
  |           ^^^^^^^

error[E0603]: module `serialize` is private
 --> tests/ui/csv_serialize_invalid_dbn_attr.rs:3:10
  |
3 | #[derive(CsvSerialize)]
  |          ^^^^^^^^^^^^
  |          |
  |          private module
  |          trait `WriteField` is not publicly re-exported
  |
note: the module `serialize` is defined here
 --> $WORKSPACE/rust/dbn/src/encode/csv.rs
  |
  | pub(crate) mod serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `write_header` found for type `u8` in the current scope
 --> tests/ui/csv_serialize_invalid_dbn_attr.rs:3:10
  |
3 | #[derive(CsvSerialize)]
  |          ^^^^^^^^^^^^ function or associated item not found in `u8`
  |
  = help: items from traits can only be used if the trait is in scope
  = note: this error originates in the derive macro `CsvSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `WriteField` which provides `write_header` is implemented but not in scope; perhaps you want to import it
  |
1 + use dbn::encode::csv::serialize::WriteField;
  |
